### DISCUSSION TITLE
How does EDC decide which interface implementation to load?

### DISCUSSION THREAD

**Discussion by mark-olchanski** [eclipse-edc/Connector#1722] [Q&A] [answered]
Hey,

if we are implementing 2 modules in our `build.gradle` files that both have an implementation of the same interface, which of those will be loaded in an extension by calling `context.getService(theInterface)`?

We are confused by this since it seems (sometimes) to depend in the order in which they get implemented in the `build.gradle` file. Is there a way to force the connector to choose one over the other?

Regards,
Mark

**Comment by jimmarino** [comment, marked as answer]
The EDC relies on the Java service loader. The correct way to handle this if you only want one implementation loaded is to ensure only it is on the classpath.
